File: front-end/src/types/Product.ts
type Product = {
  product_id: string
  shop_id: number
  product_name: string
  product_category_id: number
  product_description: string
  product_price: number
  product_stock: number
  product_details: string
  product_image_links?: string[]
}

export default Product;

File: back-end/model/product.go
package model

import "gorm.io/gorm"

type Product struct {
	gorm.Model
	ProductID          string  `json:"product_id"`
	ShopID             int     `json:"shop_id" gorm:"References:users(ID)"`
	ProductCategoryID  int     `json:"product_category_id" gorm:"References:product_categories(ProductCategoryID)"`
	ProductName        string  `json:"product_name"`
	ProductDescription string  `json:"product_description"`
	ProductPrice       float64 `json:"product_price"`
	ProductStock       int     `json:"product_stock"`
	ProductDetails     string  `json:"product_details"`
}
